tap_jira/context.py

```diff
@@ -88,9 +88,12 @@ def update_start_date_bookmark(cls, path):
 
     @classmethod
     def retrieve_timezone(cls):
-        response = cls.client.send("GET", "/rest/api/2/myself")
-        response.raise_for_status()
-        return response.json()["timeZone"]
+        # Cache the timezone to avoid repeated API calls
```

Comment:
@myleshenderson The `@cache` decorator on the function should have the same effect.

https://docs.python.org/3/library/functools.html#functools.cache
```
@cache
def factorial(n):
    return n * factorial(n-1) if n else 1

>>> factorial(10)      # no previously cached result, makes 11 recursive calls
3628800
>>> factorial(5)       # just looks up cached value result
120
>>> factorial(12)      # makes two new recursive calls, the other 10 are cached
479001600
```